src/trace_processor/perfetto_sql/generator/structured_query_generator.cc

```diff
@@ -48,6 +50,64 @@ enum QueryType : uint8_t {
   kNested,
 };
 
+std::pair<SqlSource, SqlSource> GetPreambleAndSql(const std::string& sql) {
+  PERFETTO_ELOG("Parse SQL: %s", sql.c_str());
+  std::pair<SqlSource, SqlSource> result{
+      SqlSource(SqlSource::FromTraceProcessorImplementation("")),
+      SqlSource(SqlSource::FromTraceProcessorImplementation(""))};
+
+  if (sql.empty()) {
+    return result;
+  }
+
+  SqliteTokenizer tokenizer(SqlSource::FromTraceProcessorImplementation(sql));
+
+  SqliteTokenizer::Token statement_begin_tok = tokenizer.NextNonWhitespace();
+  while (statement_begin_tok.token_type == TK_SEMI) {
```

Comment:
This still feels significantly more complex than it needs to be. Why is this not what I described in the previous one?